src/eve/concepts.py

```diff
@@ -233,53 +245,56 @@ class Config(FrozenModelConfig):
         pass
 
 
-class UIDGenerator:
-    """Simple unique id generator using a counter."""
+AnyNodeClasses = (bool, bytes, int, float, str, IntEnum, StrEnum, Node, list, dict, set)
 
-    #: Constantly increasing counter for generation of unique ids
-    __counter = itertools.count(1)
 
-    @classmethod
-    def get_unique_id(cls, *, prefix: Optional[str] = None) -> str:
-        """Generate a new globally unique id (for the current session)."""
-        count = next(cls.__counter)
-        return f"{prefix}_{count}" if prefix else f"{count}"
+class VType(FrozenModel):
+
+    # VType fields
+    #: Unique name
+    name: Str
+
+    def __init__(self, name: str) -> None:
+        super().__init__(name=name)
+
+
+# class Module:
+#     # root
 
-    @classmethod
-    def reset(cls, start: int = 1) -> None:
-        """Reset generator."""
-        cls.__counter = itertools.count(start)
+
+# class Program:
+#     # modules: List[Module]
+#     # dialects ?
 
 
-class SourceLocation(Model):
+class SourceLocation(FrozenModel):
     """Source code location (line, column, source)."""
 
     line: PositiveInt
     column: PositiveInt
     source: Str
 
+    def __init__(self, line: int, column: int, source: str) -> None:
+        super().__init__(line=line, column=column, source=source)
+
     def __str__(self) -> str:
         src = self.source or ""
         return f"<{src}: Line {self.line}, Col {self.column}>"
 
 
-class VType(Model):
-
-    # VType fields
-    #: Unique name
-    name: Str
-
-    def __init__(self, name: str) -> None:  # type: ignore
-        super().__init__(name=name)  # type: ignore
-
-    class Config(BaseModelConfig):
-        pass
-
+class UIDGenerator:
+    """Simple unique id generator using a counter."""
 
-# class Module:
-#     # root
+    #: Constantly increasing counter for generation of unique ids
```

Comment:
Proper docstring?

Reply:
Python does not support docstrings for variables (although some doc generators will emulate the feature if you define a literal string just after the declaration). Sphinx defines the `#:` syntax used here to define the docstring ins a less invasive way.

https://stackoverflow.com/questions/20227051/how-to-document-a-module-constant-in-python
https://www.sphinx-doc.org/en/1.4.8/ext/autodoc.html#directive-autoattribute